lessons/specifics/guardian.md

```diff
@@ -0,0 +1,264 @@
+---
+layout: page
+title: Guardian (Basics)
+category: specifics
+order: 7
+lang: en
+---
+
+[Guardian](https://github.com/ueberauth/guardian) is a widely used authentication library based on [JWT](https://jwt.io/) (Javascript Web Token).
+
+{% include toc.html %}
+
+## JWTs
+
+A JWT can provide a rich token for authentication. Where many authentication systems provide access to only a subject identifier for the resource, JWTs provide this along with other information like:
+
+* Who issued the token
+* Who is the token for
+* Which system should use the token
+* What time was it issued
+* What time should you consider it valid
+* What time does the token expire
+
+In addition to these fields Guardian provides some other fields to facilitate additional functionality.
+
+* What type is the token
+* What permissions does the bearer have
+
+These are just the basic fields in a JWT. You're free to add any additional information that your application requires. Just remember to keep it short. The JWT has to fit in the HTTP header.
+
+This richness means that you can pass JWTs around in your system as a fully contained unit of credentials. 
+
+### Where to use them
+
+JWT tokens can be used to authenticate any part of your application. 
+
+* Single page applications
+* Controllers (via browser session)
+* Controllers (via authoirzation headers - API)
+* Phoenix Channels
+* Service to Service requests
+* Inter-process
+* 3rd Party access (OAuth)
+* Remember me functionality
+* Other interfaces - raw TCP, UDP etc
+* Command line (if you _really_ want)
```

Comment:
This is in `specifics` (or maybe `libraries`, if we go that route) and I feel like this is going just a little too advanced for covering the dead basics. Its already made quite clear that its a portable interface, mentioning the CLI stuff is true but I think it might be unnecessary for the actual purposes of the guide.